src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/FilingsService.java

```diff
@@ -107,12 +110,13 @@ private void setFilingApiData(FilingApi filing, String requestId, String oversea
             filing.setCost(registerCostAmount);
         }
 
-        setDescriptionFields(filing);
+        setDescriptionFields(filing, isUpdateFiling);
     }
 
-    private void setSubmissionData(Map<String, Object> data, String overseasEntityId, Map<String, Object> logMap) throws SubmissionNotFoundException, ServiceException {
+    private boolean setSubmissionData(Map<String, Object> data, String overseasEntityId, Map<String, Object> logMap) throws SubmissionNotFoundException, ServiceException {
         Optional<OverseasEntitySubmissionDto> submissionOpt =
                 overseasEntitiesService.getOverseasEntitySubmission(overseasEntityId);
+        boolean isUpdateReturn = submissionOpt.isEmpty() ? false : submissionOpt.get().isForUpdate();
```

Comment:
sonar picked up a code smell for this line, you should be able to simplify it to:
`        boolean isUpdateReturn = submissionOpt.isPresent() && submissionOpt.get().isForUpdate();
`
(according to the sonar plugin in Intellij)

Reply:
fixed